File: CreditCardValidator/CreditCardValidator.java
public class CreditCardValidator{
    public static int[] convertStringDigitsToArray(String stringDigits){
        int[] arrayDigits = new int[stringDigits.length()];
        for(int count = 0; count < stringDigits.length(); count++){
            arrayDigits[count] = stringDigits.charAt(count) - '0';
        }
        return arrayDigits;
    
    }
    public static int getSumEverySecondDigitDoubled(int[] array){
        int sum = 0;
        for(int index = array.length - 2; index >= 0; index -= 2){
            if(array[index] * 2 >= 10){
                sum += ((array[index] * 2)/10 + (array[index] * 2)%10);
            }
            else{
                sum += array[index] * 2;
            }
        }
        return sum;
    
    }


    public static int getSumOfDigitsInOddIndices(int[] array){
        int sum = 0;
        for(int index = 0; index < array.length; index++){
            if(index % 2 != 0){
                sum += array[index];
            }
        }
        return sum;
    }
    
    
    
    public static int totalSum(int firstNumber, int secondNumber){
        return firstNumber + secondNumber;
    }
    
    
    public static boolean isValid(int number){
        if(number % 10 == 0){
            return true;
        }
        else{
            return false;
        }
    
    }
    
    public static String checkCardType(int[] array){
        String cardType = "";
        if(array[0] == 4){
            cardType = "Visa Card";
        }
        else if(array[0] == 5){
            cardType = "MasterCard";
        }
        else if(array[0] == 6){
            cardType = "Discover Card";     
        }
        else if(array[0] == 3 && array[1] == 7){
            cardType = "American Express Card";
        }
        else{
            cardType = "Invalid Card";
        }
        return cardType;
   
    } 
    
    public static String checkDigitSize(int[] array){
        if(array.length < 13){
            return "Invalid Card";
        }
        else{
            return "Valid Card";
        }
    }

}

File: CreditCardValidator/Python/credit_card_validator.py
def get_sum_every_second_digit_doubled(string_digits):
    total_sum = 0
    index = len(string_digits) - 2
    while index >= 0:
        doubled = int(string_digits[index]) * 2
        if doubled >= 10:
            total_sum += ((doubled // 10) + (doubled % 10))
        else:
            total_sum += doubled           
        index -= 2
    return total_sum


def get_sum_of_digits_in_odd_indices(string_digits):
    total_odd_sum = 0
    index = len(string_digits) - 1
    while index >= 0:
        digit = int(string_digits[index])
        total_odd_sum += digit     
        index -= 2    
    return total_odd_sum


def total_sum(first_number, second_number):
    return first_number + second_number


def get_card_validity(string_digits, number):
    if len(string_digits) < 13 or len(string_digits) > 16:
        return "Invalid"
    if number % 10 == 0:
        return "Valid"
    else:
        return "Invalid"

def check_card_type(string_digits):
    if len(string_digits) == 0:
        return "Invalid Card Type"
    if string_digits[0] == "4":
        return "Visa Card"
    elif string_digits[0] == "5":
        return "MasterCard"
    elif string_digits[0] == "6":
        return "Discover Card"
    elif len(string_digits) > 1 and string_digits[0] == "3" and string_digits[1] == "7":
        return "American Express Card"
    else:
        return "Invalid Card Type"



